src/hierarchy.ts

```diff
@@ -220,7 +217,7 @@ export class Group implements AsyncMutableMapping<Group | ZarrArray> {
     async getItem(item: string) {
         const path = this.itemPath(item);
         if (await containsArray(this.store, path)) {
-            return ZarrArray.create(this.store, this.path, this.readOnly, this.chunkStore, undefined, this.attrs.cache);
+            return ZarrArray.create(this.store, path, this.readOnly, this.chunkStore, undefined, this.attrs.cache);
```

Comment:
This was a bug. It would try to create an array at the group's path, not the normalized path from `this.itemPath`.